templates/deployment.yaml

```diff
@@ -233,3 +237,178 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+
+# Netbox Worker
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "netbox.fullname" . }}-worker
+  labels:
+    {{- include "netbox.labels" . | nindent 4 }}
+spec:
+{{- if not .Values.worker.autoscaling.enabled }}
+  replicas: {{ .Values.worker.replicaCount }}
+{{- end }}
+  selector:
+    matchLabels:
+      {{- include "netbox.selectorLabels" . | nindent 6 }}
+  {{ if .Values.worker.updateStrategy -}}
+  strategy:
+    {{- toYaml .Values.worker.updateStrategy | nindent 4 }}
+  {{ end -}}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        {{- with .Values.worker.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "netbox.selectorLabels" . | nindent 8 }}
```

Comment:
Currently the worker ends up with the same labels as the web service. This would lead to confusion with the deployments and services, in particular the Service would have endpoints pointing at the worker pods. We'll need to add labels both here and to the main web service deployment, and make the relevant changes to the Service and HPA. The changes in selectors will be a breaking change for the chart, too, so we'll need a major version bump and a suitable explanation and upgrade instructions in the README, please.

Reply:
Ah, I see, the Selectors should be changed.

But I'm quite new to k8s & helm-charts, therefore I don't think I quite get it yet:
- the Service uses Selectors to determine its backend
  - Selectors are defined centrally in `_helpers.tpl`
- the Selectors, that will be matched, are defined on the main Deployment as well

Would it not be sufficient to just define different Selectorlabels for worker Deployment?

Then there would not be a conflict between main and worker Deployment, as I understand it, and therefore no breaking of current functionality.

E.g. extend `_helpers.tpl` by "netbox.worker.selectorLabels":
```yaml
{{/*
Worker Selector labels
*/}}
{{- define "netbox.selectorLabels" -}}
app.kubernetes.io/name: {{ include "netbox.name" . }}-worker
app.kubernetes.io/instance: {{ .Release.Name }}
{{- end }}
```

And then define them on worker HPA & Deployment.

Reply:
Well, it's more that there are conventions around the selector labels; please see https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/. We should really use `app.kubernetes.io/component` to denote the actual NetBox component vs the rqworker.

It turns out that we'll likely need a major version bump for NetBox 2.10 anyway, so I'll merge this manually and fix up the selector labels.